pkgs/by-name/ni/nixos-rebuild-ng/src/nixos_rebuild/models.py

```diff
@@ -57,7 +58,30 @@ def to_attr(self, *attrs: str) -> str:
     @classmethod
     def from_arg(cls, attr: str | None, file: str | None) -> Self:
         if not (attr or file):
-            return cls("<nixpkgs/nixos>", None)
+            # If present, use config.nixpkgs.source
+            nixos_config = os.getenv("NIXOS_CONFIG", "<nixos-config>")
+            r = run_wrapper(
+                [
+                    "nix-instantiate",
+                    "--eval",
+                    "--expr",
+                    f"""
+                      (import <nixpkgs/nixos> {{
+                        configuration = {{
+                          imports = [ {nixos_config} ];
+                          _module.check = false;
+                        }};
+                      }}).config.nixpkgs.source
+                    """,
```

Comment:
Nit, but probably applying `textwrap.dedent` is a good idea here, otherwise it will unnecessary pass a very long string (and it will especially be bad in the logs if enabled).

Reply:
Sure, I had noticed the "\n                           \n                 " in the debug log. Done.